runtime/dart_snapshot.cc

```diff
@@ -202,4 +204,16 @@ const uint8_t* DartSnapshot::GetInstructionsMapping() const {
   return instructions_ ? instructions_->GetMapping() : nullptr;
 }
 
+bool DartSnapshot::IsNullSafetyEnabled(const fml::Mapping* kernel) const {
```

Comment:
Earlier you used `std::optional`. How are you choosing between that and this?

Reply:
IIRC, earlier it was an optional vector, now its an pointer that may be null. I could make it an optional pointer or an optional reference. A pointer just seemed cleaner here.